src/utils.c

```diff
@@ -121,3 +121,43 @@ void header_meta_from_first(FILE *mib_ptr,
       current_file, __LINE__);
   }
 }
+
+hid_t bufsize_to_datatype(int dtype)
+{
+  hid_t datatype;
+  switch (dtype) {
+    case 1: {
+      datatype = H5T_STD_U8LE;
+      break;
+    }
+    case 2: {
+      datatype = H5T_STD_U16LE;
+      break;
+    }
+    case 4: {
+      datatype = H5T_STD_U32LE;
+      break;
+    }
+    case 8: {
+      datatype = H5T_STD_U64LE;
+      break;
+    }
+    default: {
+      fprintf(stderr, "Error in datatype, please check input dtype\n");
+      return H5I_INVALID_HID;
+    }
+  }
+  return datatype;
+}
+
+unsigned long get_filesystem_block_size(const char *path)
+{
+  struct statvfs stat;
```

Comment:
While looking up this out of interests, the [example in the man page](https://man7.org/linux/man-pages/man3/statvfs.3.html) will `#include <sys/statvfs.h>`. Probably this header is included somewhere so its declaration can be found, but I would suggest we include this header in this file as well.